### PR TITLE
Add `load-partial-extension` and deprecate `at-import-partial-extension`.

### LINKED ISSUE #959: Extend `at-import-partial-extension` to `load-partial-extension`
Enhancement to [at-import-partial-extension](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/at-import-partial-extension/README.md) so it applies to `@use`, `@forward`, and [meta.load-css()](https://sass-lang.com/documentation/modules/meta/#mixins) as well.

### PR DESCRIPTION
**Pull request by pamelalozano16**
Fix #959.

### CODE REVIEW

**Comment by kristerkari** [comment]
Thanks! I think that this good to go after I just read through the changes 👍

**Comment by pamelalozano16** [comment]
Hi @kristerkari! Any thoughts on these changes ? 😄

**Comment by kristerkari** [comment]
@pamelalozano16 Sorry about the delay. I'll review this now.

File: src/rules/at-import-partial-extension/index.js [hunk 1]
```diff
@@ -47,6 +48,16 @@ function rule(expectation, _, context) {
       return;
     }
 
+    result.warn(
+      "'at-import-partial-extension has been deprecated, " +
+        "and will be removed in '7.0'. Use 'load-partial-extensione' instead.",
```
Comment (kristerkari) [inline src/rules/at-import-partial-extension/index.js:53]: `load-partial-extensione` <- typo

File: src/rules/load-partial-extension/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+ * This extension */
+```
+
+```scss
+@use "sass:meta";
+
+body {
+  @include meta.load-css("fff.scss",
+                  /**         ↑
+                  * This extension */
```
Comment (kristerkari) [inline src/rules/load-partial-extension/README.md:17]: Small thing, but the indentation of this comment is a bit off. 😄

File: src/rules/load-partial-extension/index.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    if (!validOptions) {
+      return;
+    }
+
+    root.walkAtRules(atRule => {
+      if (
+        atRule.name === "import" ||
+        atRule.name === "use" ||
+        atRule.name === "forward" ||
+        atRule.name === "include"
```
Comment (kristerkari) [inline src/rules/load-partial-extension/index.js:58]: Small detail, but the code could be a bit easier to read by doing something like this:

```js
const loadAtRules = ["import", "use", "forward", "include"];

// https://drafts.csswg.org/mediaqueries/#media-types
const mediaQueryTypes = [
```

---

```js
    root.walkAtRules(atRule => {
      if (loadAtRules.includes(atRule.name)) {

```

File: src/rules/load-partial-extension/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      ) {
+        let name = atRule.name;
+        let paths = atRule.params
+          .split(/["']\s*,/)
+          .filter(path => !mediaQueryTypesRE.test(path.trim()));
+
+        if (atRule.params.match(/load-css/)) {
```
Comment (kristerkari) [inline src/rules/load-partial-extension/index.js:65]: Could this instead be:

```js
        const isLoadCss = !!atRule.params.match(/load-css/);

        if (isLoadCss) {
          name = "meta.load-css";
```

**Comment by kristerkari** [comment]
@pamelalozano16 I left a few small comments, other than that looks good 👍

**Comment by pamelalozano16** [comment]
@kristerkari Thanks for the review! Should be ready now to merge. :)

**Comment by kristerkari** [comment]
Thanks, looks good now 👍

**Pull request merged**